### ISSUE TITLE
BatchNormalization中的training参数

### ISSUE THREAD

**Issue by wangzhe258369** [ZiyaoGeng/RecLearn#4]
https://github.com/ZiyaoGeng/Recommender-System-with-TF2.0/blob/90c2101917f92142b05ca88732e39e542647b46b/DIN/dice.py#L20

这里需要：
```
    def call(self, x, training=True):
        x_normed = self.bn(x, training=training)
```

否则你就被TensorFlow的batch normalization实现坑啦。

**Comment by ZiyaoGeng** [comment]
要指定trainable么？感谢指出，之前并不清楚

**Comment by wangzhe258369** [comment]
是training，不是trainable。

tf.keras.layers.BatchNormalization默认行为是training=False，此时不会去更新BN中的moving_mean和moving_variance变量，相当于这两个变量随机初始化后就一直作为一个“常量”在用。

**Comment by ZiyaoGeng** [comment]
去看了下官方文档，发现确实是这样，之前忽略了，非常感谢！



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Wangzhe"<notifications@github.com&gt;; 
发送时间: 2020年9月10日(星期四) 上午10:14
收件人: "ZiyaoGeng/Recommender-System-with-TF2.0"<Recommender-System-with-TF2.0@noreply.github.com&gt;; 
抄送: "艹(≧▽≦)八千流"<593947521@qq.com&gt;; "Comment"<comment@noreply.github.com&gt;; 
主题: Re: [ZiyaoGeng/Recommender-System-with-TF2.0] BatchNormalization中的training参数 (#4)





 
是training，不是trainable。
 
tf.keras.layers.BatchNormalization默认行为是training=False，此时不会去更新BN中的moving_mean和moving_variance变量，相当于这两个变量随机初始化后就一直作为一个“常量”在用。

**Comment by ZiyaoGeng** [comment]
重写了DIN模型，BN查了一下资料，现在是不用指定traning参数的，具体的可以看下https://www.it610.com/article/1276108622954250240.htm
